field/src/helpers.rs

```diff
@@ -130,32 +130,33 @@ pub fn halve_u64<const P: u64>(input: u64) -> u64 {
     }
 }
 
-/// Given a slice of SF elements, reduce them to a TF element.
-pub fn reduce_64<SF: PrimeField64, TF: PrimeField>(vals: &[SF]) -> TF {
-    let alpha = TF::from_canonical_u64(SF::ORDER_U64);
-
-    let mut res = TF::zero();
-    for val in vals.iter().rev() {
-        res = res * alpha + TF::from_canonical_u64(val.as_canonical_u64());
+/// Given a slice of SF elements, reduce them to a TF element using a 2^32-base decomposition.
+pub fn reduce_32<SF: PrimeField32, TF: PrimeField>(vals: &[SF]) -> TF {
+    let po2 = TF::from_canonical_usize(1usize << 32);
+    let mut power = TF::one();
+    let mut result = TF::zero();
+    for val in vals {
+        result += TF::from_canonical_u32(val.as_canonical_u32()) * power;
+        power *= po2;
```

Comment:
Not sure if performance really matters here, but the previous Horner-style method involving `rev()` seems a bit more nicer for a single mul per iteration, is there a downside to that?